src/code/z_message.c

```diff
@@ -64,46 +322,239 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_801514B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/Message_StartTextbox.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151938.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151A68.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151BB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151C9C.s")
+void Message_StartTextbox(GlobalContext* globalCtx, u16 textId, Actor* Actor) {
+    MessageContext* msgCtx;
+    msgCtx = &globalCtx->msgCtx;
+    msgCtx->ocarinaAction = 0xFFFF;
+    func_80150D08(globalCtx, textId);
+    msgCtx->unkActor = Actor;
+    msgCtx->msgMode = 1;
+    msgCtx->unk12023 = 0;
+    msgCtx->unk12024 = 0;
+    globalCtx->msgCtx.ocarinaMode = 0;
+}
+
+void func_80151938(GlobalContext* globalCtx, u16 textId) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+
+    msgCtx->unk11F10 = 0;
+    func_80150D08(globalCtx, textId);
+    func_80150A84(globalCtx);
+    msgCtx->msgMode = 5;
+    msgCtx->unk12023 = 8;
+    msgCtx->unk12024 = 0;
+
+    if (interfaceCtx->unk_222 == 0) {
+        if (textId != 0x1B93) {
+            func_8011552C(globalCtx, 0x10);
+        } else if (textId != 0xF8) {
+            func_8011552C(globalCtx, 6);
+        }
+    }
+    msgCtx->unk1203C = msgCtx->unk1203A;
+    if (globalCtx->pauseCtx.unk_1F0 != 0) {
+        msgCtx->unk12004 = 0x22;
+        msgCtx->unk12006 = 0x15E;
+        func_80149C18(globalCtx);
+        msgCtx->unk12023 = 1;
+    }
+}
+
+void func_80151A68(GlobalContext* globalCtx, u16 textId) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+
+    msgCtx->unk11F10 = 0;
+    func_80150D08(globalCtx, textId);
+    func_80150A84(globalCtx);
+    func_8015B198(globalCtx);
+    msgCtx->msgMode = 0x45;
+    msgCtx->unk12024 = 0;
+    msgCtx->unk1203C = (msgCtx->unk1203A = (msgCtx->unk1201E = 0));
+    msgCtx->unk12023 = 0x1E;
+
+    // Day/Dawn/Night.. Messages
+    if (((msgCtx->currentTextId) >= 0x1BB2) && ((msgCtx->currentTextId) < 0x1BB7)) {
```

Comment:
```suggestion
    if ((msgCtx->currentTextId >= 0x1BB2) && (msgCtx->currentTextId < 0x1BB7)) {
```

Reply:
done in https://github.com/zeldaret/mm/commit/eaa4f44462da18c9c81ba828fca65470abf3b526